<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>单细胞数据整合：读Paper/翻Github时发现的一些工具 | WordsOverFlow:(</title><meta name="keywords" content="单细胞数据整合,单细胞工具"><meta name="author" content="Arita Zhang"><meta name="copyright" content="Arita Zhang"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="ATTENTION: 本文的内容将持续更新。  总结一下近期读paper&#x2F;翻Github时发现的一些可以用于做单细胞数据整合 的工具，大致分为去除批次效应 、单模态数据整合 、多模态数据整合 几个大类，一些能用在单一模态（如scRNA-seq）上的工具因为我本人没用过不清楚能不能迁移到其他模态上（如染色质可及性、DNA甲基化等），在适用范围上也就先不标了。 Paper主要参考的是Luecken等人">
<meta property="og:type" content="article">
<meta property="og:title" content="单细胞数据整合：读Paper&#x2F;翻Github时发现的一些工具">
<meta property="og:url" content="https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/index.html">
<meta property="og:site_name" content="WordsOverFlow:(">
<meta property="og:description" content="ATTENTION: 本文的内容将持续更新。  总结一下近期读paper&#x2F;翻Github时发现的一些可以用于做单细胞数据整合 的工具，大致分为去除批次效应 、单模态数据整合 、多模态数据整合 几个大类，一些能用在单一模态（如scRNA-seq）上的工具因为我本人没用过不清楚能不能迁移到其他模态上（如染色质可及性、DNA甲基化等），在适用范围上也就先不标了。 Paper主要参考的是Luecken等人">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/%E5%8D%95%E7%BB%86%E8%83%9E%E6%95%B4%E5%90%88%E7%AD%96%E7%95%A5.png">
<meta property="article:published_time" content="2022-01-12T06:55:17.000Z">
<meta property="article:modified_time" content="2022-01-15T04:11:11.830Z">
<meta property="article:author" content="Arita Zhang">
<meta property="article:tag" content="单细胞数据整合">
<meta property="article:tag" content="单细胞工具">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/%E5%8D%95%E7%BB%86%E8%83%9E%E6%95%B4%E5%90%88%E7%AD%96%E7%95%A5.png"><link rel="shortcut icon" href="/img/logo2.png"><link rel="canonical" href="https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  date_suffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: undefined,
  lightbox: 'fancybox',
  Snackbar: undefined,
  source: {
    jQuery: 'https://cdn.jsdelivr.net/npm/jquery@latest/dist/jquery.min.js',
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/js/jquery.justifiedGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/justifiedGallery/dist/css/justifiedGallery.min.css'
    },
    fancybox: {
      js: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.js',
      css: 'https://cdn.jsdelivr.net/npm/@fancyapps/fancybox@latest/dist/jquery.fancybox.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isanchor: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: '单细胞数据整合：读Paper/翻Github时发现的一些工具',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2022-01-15 12:11:11'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(win=>{
    win.saveToLocal = {
      set: function setWithExpiry(key, value, ttl) {
        if (ttl === 0) return
        const now = new Date()
        const expiryDay = ttl * 86400000
        const item = {
          value: value,
          expiry: now.getTime() + expiryDay,
        }
        localStorage.setItem(key, JSON.stringify(item))
      },

      get: function getWithExpiry(key) {
        const itemStr = localStorage.getItem(key)

        if (!itemStr) {
          return undefined
        }
        const item = JSON.parse(itemStr)
        const now = new Date()

        if (now.getTime() > item.expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return item.value
      }
    }
  
    win.getScript = url => new Promise((resolve, reject) => {
      const script = document.createElement('script')
      script.src = url
      script.async = true
      script.onerror = reject
      script.onload = script.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        script.onload = script.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(script)
    })
  
      win.activateDarkMode = function () {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = function () {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
          if (t === 'dark') activateDarkMode()
          else if (t === 'light') activateLightMode()
        
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
    const detectApple = () => {
      if (GLOBAL_CONFIG_SITE.isHome && /iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
    })(window)</script><meta name="generator" content="Hexo 5.4.0"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="/img/avatar2.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="site-data"><div class="data-item is-center"><div class="data-item-link"><a href="/archives/"><div class="headline">文章</div><div class="length-num">3</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/tags/"><div class="headline">标签</div><div class="length-num">3</div></a></div></div><div class="data-item is-center"><div class="data-item-link"><a href="/categories/"><div class="headline">分类</div><div class="length-num">1</div></a></div></div></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友情链接</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于</span></a></div><div class="menus_item"><a class="site-page" href="/shuoshuo/"><i class="fa-fw fa fa-cubes"></i><span> 自说自话</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="not-top-img" id="page-header"><nav id="nav"><span id="blog_name"><a id="site-name" href="/">WordsOverFlow:(</a></span><div id="menus"><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 主页</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 时间轴</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 标签</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分类</span></a></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友情链接</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 关于</span></a></div><div class="menus_item"><a class="site-page" href="/shuoshuo/"><i class="fa-fw fa fa-cubes"></i><span> 自说自话</span></a></div></div><div id="toggle-menu"><a class="site-page"><i class="fas fa-bars fa-fw"></i></a></div></div></nav></header><main class="layout" id="content-inner"><div id="post"><div id="post-info"><h1 class="post-title">单细胞数据整合：读Paper/翻Github时发现的一些工具</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2022-01-12T06:55:17.000Z" title="发表于 2022-01-12 14:55:17">2022-01-12</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2022-01-15T04:11:11.830Z" title="更新于 2022-01-15 12:11:11">2022-01-15</time></span><span class="post-meta-categories"><span class="post-meta-separator">|</span><i class="fas fa-inbox fa-fw post-meta-icon"></i><a class="post-meta-categories" href="/categories/%E5%8D%95%E7%BB%86%E8%83%9E%E5%88%86%E6%9E%90/">单细胞分析</a></span></div><div class="meta-secondline"></div></div></div><article class="post-content" id="article-container"><mark class="hl-label red">ATTENTION:</mark> 本文的内容将持续更新。

<p>总结一下近期读paper/翻Github时发现的一些可以用于做<mark class="hl-label red">单细胞数据整合</mark> 的工具，大致分为<mark class="hl-label blue">去除批次效应</mark> 、<mark class="hl-label pink">单模态数据整合</mark> 、<mark class="hl-label orange">多模态数据整合</mark> 几个大类，一些能用在单一模态（如scRNA-seq）上的工具因为我本人没用过不清楚能不能迁移到其他模态上（如染色质可及性、DNA甲基化等），在适用范围上也就先不标了。</p>
<p>Paper主要参考的是Luecken等人22年发在Nature Method上的Benchmarking、Argelaguet等人21年发在Nature Biotechnology上的综述，以及各个工具的作者在其文章Introduction &amp; Results部分提及的其他工具的算法原理与性能情况。</p>
<p>然后再贴一下<mark class="hl-label purple">awesome-single-cell</mark> 这个Github Repository，单细胞数据分析软件贴的比较全，整合算法因为最近一两年出来的比较多上面还没怎么更新。<a target="_blank" rel="noopener" href="https://github.com/seandavi/awesome-single-cell">Github Link</a>。</p>
<mark class="hl-label default">WARNING:</mark> 因为本人对算法原理都不是很熟悉，且数学知识早就还给教过我的各位老师了，尽管描述算法的部分写的都比较简陋，还是看看就好OTL

<ul>
<li><h2 id="scvi-tools"><a href="#scvi-tools" class="headerlink" title="scvi-tools"></a><a target="_blank" rel="noopener" href="https://docs.scvi-tools.org/en/stable/index.html">scvi-tools</a></h2><ol>
<li>18年至今陆陆续续开发的一组工具包，应用神经网络基于潜在变量模型（Latent Variable models）开发深度生成模型（DGMs）解决诸如降维、聚类、差异表达分析、注释、去除不想要的变异（去批次效应） &amp; 跨模态整合的问题。该套工具包包括可分析scRNA-seq的工具（scVI、scArches），单细胞注释（CellAssign, scANVI），空间转录组数据的去卷积化（Stereoscope, DestVI），去除双细胞（Solo），基于CITE-seq数据的多模态分析（totalVI），染色质可及性分析（PeakVI）等。</li>
<li>scvi-tools利用AnnData数据结构，可和Scanpy集成；同时也具有R接口（reticulate），在一定程度上可以和Seurat/Bioconductor协同工作。</li>
<li>利用神经网络进行训练需要一定的电脑配置（无现成GPU资源的情况下推荐用Google的colab白嫖GPU），大数据（几十k细胞）用CPU训练模型极其缓慢。</li>
</ol>
</li>
<li><h2 id="Scanorama"><a href="#Scanorama" class="headerlink" title="Scanorama"></a><a target="_blank" rel="noopener" href="http://cb.csail.mit.edu/cb/scanorama/">Scanorama</a></h2><ol>
<li>18年发布的一款可去除scRNA-seq批次效应 &amp; 整合数据的算法，利用全景拼接方法，根据两个数据集Overlap的区域进行batch-correction和Integration，能够保留数据 specific population，也不需要所有的dataset share相同的cell type。</li>
<li>利用mutual nearest neighbors matching ，主要能够寻找两个数据集的相似部分，而不是整个大数据中的相似部分。针对多于两个datasets的比对，先选择一个作为reference，之后将其他的datasets和这个datasets做integrate，所以整合的效果很看重数据的顺序。</li>
<li>Scanorama 的两个关键性步骤：1.使用SVD（singular value decomposition） 将高维的数据降维。2.基于超敏感局部敏感散列和随机投影树来构建近邻法，节约时间。</li>
<li>主要适用于scRNA-seq的批次效应去除 &amp; 整合，在Argelaguet的综述中被归类于<mark class="hl-label green">Horizontal-Integration</mark> 。</li>
<li>Python package，利用AnnData作为数据结构，可集成Scanpy。</li>
</ol>
</li>
<li><h2 id="fastMNN"><a href="#fastMNN" class="headerlink" title="fastMNN"></a>fastMNN</h2></li>
<li><h2 id="scGen"><a href="#scGen" class="headerlink" title="scGen"></a><a target="_blank" rel="noopener" href="https://scgen.readthedocs.io/en/latest/">scGen</a></h2><ol>
<li>基于结合向量算法的VAE，具有适合scRNA-seq数据的体系结构。scGen能够预测细胞在药剂或感染下的变化，以预测细胞类型；且scGen能够跨物种的挖掘细胞受扰动产生的表达改变。</li>
<li>（从scGen的原始论文来看），scGen适用于scRNA-seq对照组-实验组的去批次效应和整合，也适合做跨物种分析，是否适用于更广范围的去批次效应 &amp; scRNA-seq整合需要进一步实验。</li>
<li>Python package, 利用AnnData为数据结构，也可集成Scanpy（scGen的去批次效应和分析需要scanpy的<code>highly_variable_gene</code>函数。</li>
</ol>
</li>
<li><h2 id="Harmony"><a href="#Harmony" class="headerlink" title="Harmony"></a>Harmony</h2></li>
<li><h2 id="Seurat-v3"><a href="#Seurat-v3" class="headerlink" title="Seurat v3"></a>Seurat v3</h2><ol>
<li>老牌scRNA-seq分析工具，v3版本似乎可直接根据scATAC-seq的染色质峰矩阵计算Gene Activity进行scRNA-seq &amp; scATAC-seq的整合，在Argelaguet的综述中被归类为<mark class="hl-label green">Horizontal-Integration</mark> 。</li>
<li>虽然没有用过，但在Leucken的Benchmarking文章中去批次效应结果&amp;整合结果都表现欠佳。总之和它的后续版本Seurat v4一样，如果能选择基于深度学习的工具，那就不要用它做整合。</li>
<li>这东西是个R包就不用说了吧。</li>
</ol>
</li>
<li><h2 id="Seurat-v4"><a href="#Seurat-v4" class="headerlink" title="Seurat v4"></a><a target="_blank" rel="noopener" href="https://satijalab.org/seurat/index.html">Seurat v4</a></h2><ol>
<li>老牌scRNA-seq分析工具，v4版本在代码风格上和v3版本有一定差异；该版本结合了该Satija实验室开发的Signac工具，可将scATAC-seq的fragments数据转化为基因活度（Gene Activity）。v4版本的Seurat利用CCA算法将基因表达 &amp; 基因活度降维后寻找两种模态中通用的anchor进行scRNA-seq -&gt; scATAC-seq的映射，从而完成两种模态的整合。</li>
<li>集成了WNN算法，对单细胞多模态数据（<strong>profiled in same cells</strong>）的矫正&amp;整合力度似乎还不错，在Argelaguet的综述中被归类为<mark class="hl-label green">Vertical-Integration</mark> 。</li>
<li>只能进行scRNA-seq和scATAC-seq的整合；采用的anchors算法在挖掘两种模态的通用特征上不够深入；将染色质可及性的峰矩阵转换为基因活度矩阵会丢失一部分原数据的特性，进一步影响效果。</li>
<li>多个比较实验中，Seurat v4在标签转移、多模态数据整合上表现欠佳（见Leucken/UnionCom/SCOTv2文章）。&amp; 一点个人经验：如果能用基于深度学习的工具（SCOT、scVI、UnionCom等），不要用Seurat v4。</li>
<li>这东西是个R包就不用说了吧。</li>
</ol>
</li>
<li><h2 id="BBKNN"><a href="#BBKNN" class="headerlink" title="BBKNN"></a>BBKNN</h2></li>
<li><h2 id="ComBat"><a href="#ComBat" class="headerlink" title="ComBat"></a>ComBat</h2></li>
<li><h2 id="MNN"><a href="#MNN" class="headerlink" title="MNN"></a>MNN</h2></li>
<li><h2 id="trVAE"><a href="#trVAE" class="headerlink" title="trVAE"></a>trVAE</h2></li>
<li><h2 id="Conos"><a href="#Conos" class="headerlink" title="Conos"></a>Conos</h2></li>
<li><h2 id="DESC"><a href="#DESC" class="headerlink" title="DESC"></a>DESC</h2></li>
<li><h2 id="LIGER"><a href="#LIGER" class="headerlink" title="LIGER"></a>LIGER</h2></li>
<li><h2 id="SAUCIE"><a href="#SAUCIE" class="headerlink" title="SAUCIE"></a>SAUCIE</h2></li>
<li><h2 id="scAlign"><a href="#scAlign" class="headerlink" title="scAlign"></a>scAlign</h2></li>
<li><h2 id="MATCHER"><a href="#MATCHER" class="headerlink" title="MATCHER"></a>MATCHER</h2></li>
<li><h2 id="MAGAN"><a href="#MAGAN" class="headerlink" title="MAGAN"></a>MAGAN</h2></li>
<li><h2 id="Harmonic"><a href="#Harmonic" class="headerlink" title="Harmonic"></a>Harmonic</h2></li>
<li><h2 id="MMD-MA"><a href="#MMD-MA" class="headerlink" title="MMD-MA"></a>MMD-MA</h2></li>
<li><h2 id="SCIM"><a href="#SCIM" class="headerlink" title="SCIM"></a>SCIM</h2><ol>
<li>SCIM假设细胞共享一个共同的（低维的）底层结构，并且底层细胞的分布在不同的技术中心是近似恒定的，利用一个具有对抗性目标的自动编码器框架构造了一个技术不变的潜在空间。多模态数据集市通过配对跨技术的单元来集成的，使用一个二部匹配方案，操作在低维潜在表示。</li>
<li>主要应用于分别测定的单细胞多模态数据（<strong>profiled in different cells</strong>）的整合，在Argelaguet的综述中被归类为<mark class="hl-label green">Diagonal-Integration</mark> 。</li>
</ol>
</li>
<li><h2 id="UnionCom"><a href="#UnionCom" class="headerlink" title="UnionCom"></a><a target="_blank" rel="noopener" href="https://github.com/caokai1073/UnionCom">UnionCom</a></h2><ol>
<li>一种基于流型策略，利用了拓扑比对（Topological Alignment）的无监督算法。UnionCom将每个单细胞数据集固有的低维结构嵌入到几何距离矩阵中，通过匹配几何距离矩阵，跨单细胞多组学数据集对齐细胞，最终将不同的不对应特征投射到一个共同的embedding空间中。</li>
<li>主要应用于分别测定的单细胞多模态数据(<strong>profiled in different cells</strong>)的整合，<mark class="hl-label green">Diagonal-Integration</mark> 。</li>
<li>不需要事先的细胞关联信息作为输入（不需要各模态所检测细胞的1-1匹配，也不需要事先的Cell Type Labels）；采用非线性方法，在非线性特征（如染色质可及性上）较线性方法如Seurat/MNN/Harmony等有更好的性能。</li>
<li>是<strong>Python</strong>工具包，可以在代码框中运行，不依赖Linux命令行。</li>
<li>基于UnionCom，作者Cao Kai开发了<a target="_blank" rel="noopener" href="https://github.com/caokai1073/Pamona">Panoma</a>，可集成细胞类型的先验信息。</li>
</ol>
</li>
<li><h2 id="coupledNMF"><a href="#coupledNMF" class="headerlink" title="coupledNMF"></a>coupledNMF</h2></li>
<li><h2 id="SCOTv2"><a href="#SCOTv2" class="headerlink" title="SCOTv2"></a><a target="_blank" rel="noopener" href="https://github.com/rsinghlab/SCOT">SCOTv2</a></h2><ol>
<li>一种基于不平衡最佳传输（Unbalanced Optimal transport）策略的无监督算法，该v2版本算法在SCOT（采用Gromov-Wasserstein最佳传输算法）的基础上增加了散度项，以允许边缘值的质量变化。为允许更多模态（N&gt;2）的整合，使用分类更为清晰质量更好的domain（我猜测可能是整个模态的数据）作为anchor进行不同模态间的锚定。SCOTv2采用了UnionCom利用的调整后的tsne，基于偶联的细胞-细胞转移矩阵（cell-cell transition matrix）进行coembedding的展示。</li>
<li>主要应用于分别测定的单细胞多模态数据（<strong>profiled in different cells</strong>）的整合，可适用于细胞类型不平衡、缺乏1-1细胞关联的情况，属于<mark class="hl-label green">Diagonal-Integration</mark> 。</li>
<li>该算法可以在构建模型的过程中自动选取最优参数。</li>
<li>可一次性整合多个模态（N&gt;2，例如可一次性整合scNMT-seq的基因表达、染色质可及性、DNA甲基化到一张图中），且不需要事先的Cell Type Labels暨分类依据作为输入。</li>
<li>可直接在CPU上运行，在多种算法中运行速度较快；但其对内存的需求为$O(N^2)$，可能不适用于大规模的数据。</li>
<li>是<strong>Python</strong>工具包，可以在代码框中运行，不依赖Linux命令行（这篇文章中使用的电脑配置简直就是贫民款）。</li>
</ol>
</li>
<li><h2 id="MOFA"><a href="#MOFA" class="headerlink" title="MOFA+"></a>MOFA+</h2></li>
<li><h2 id="scmap"><a href="#scmap" class="headerlink" title="scmap"></a>scmap</h2></li>
<li><h2 id="BindSC"><a href="#BindSC" class="headerlink" title="BindSC"></a>BindSC</h2><ol>
<li>通过将scATAC-seq的峰矩阵转化为基因活度矩阵，进行scRNA-seq &amp; scATAC-seq的整合。</li>
<li>在SCOTv2的论文中，在label transfer &amp; integration embedding上表现不佳，如果条件允许还是建议选择其他的工具进行整合。</li>
</ol>
</li>
</ul>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="mailto:undefined">Arita Zhang</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/">https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="https://AritaZ-hang.github.io" target="_blank">WordsOverFlow:(</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/%E5%8D%95%E7%BB%86%E8%83%9E%E6%95%B0%E6%8D%AE%E6%95%B4%E5%90%88/">单细胞数据整合</a><a class="post-meta__tags" href="/tags/%E5%8D%95%E7%BB%86%E8%83%9E%E5%B7%A5%E5%85%B7/">单细胞工具</a></div><div class="post_share"><div class="social-share" data-image="/2022/01/12/single-cell-multiomics-integration/%E5%8D%95%E7%BB%86%E8%83%9E%E6%95%B4%E5%90%88%E7%AD%96%E7%95%A5.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/social-share.js/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/social-share.js/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2022/01/15/scvi-tools-trap/"><img class="prev-cover" src="/2022/01/15/scvi-tools-trap/scvi-tools.jpg" onerror="onerror=null;src='/img/404.jpg'" alt="cover of previous post"><div class="pagination-info"><div class="label">上一篇</div><div class="prev_info">记在用scvi-tools时踩过的坑</div></div></a></div><div class="next-post pull-right"><a href="/2022/01/12/biostars-first-try/"><img class="next-cover" src="/2022/01/12/biostars-first-try/biostars.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">单细胞多模态整合：记我在Biostars上的第一次提问</div></div></a></div></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>相关推荐</span></div><div class="relatedPosts-list"><div><a href="/2022/01/15/scvi-tools-trap/" title="记在用scvi-tools时踩过的坑"><img class="cover" src="/2022/01/15/scvi-tools-trap/scvi-tools.jpg" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2022-01-15</div><div class="title">记在用scvi-tools时踩过的坑</div></div></a></div><div><a href="/2022/01/12/biostars-first-try/" title="单细胞多模态整合：记我在Biostars上的第一次提问"><img class="cover" src="/2022/01/12/biostars-first-try/biostars.png" alt="cover"><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2022-01-12</div><div class="title">单细胞多模态整合：记我在Biostars上的第一次提问</div></div></a></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="is-center"><div class="avatar-img"><img src="/img/avatar2.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info__name">Arita Zhang</div><div class="author-info__description">-流而不盈 持中守正-</div></div><div class="card-info-data"><div class="card-info-data-item is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">3</div></a></div><div class="card-info-data-item is-center"><a href="/tags/"><div class="headline">标签</div><div class="length-num">3</div></a></div><div class="card-info-data-item is-center"><a href="/categories/"><div class="headline">分类</div><div class="length-num">1</div></a></div></div><a class="button--animated" id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/AritaZ-hang"><i class="fab fa-github"></i><span>Follow Me(๑•̀ㅂ•́)و✧</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://github.com/AritaZ-hang" target="_blank" title="Github"><i class="fab fa-github"></i></a><a class="social-icon" href="mailto:cosmos_clarify@foxmail.com" target="_blank" title="Email"><i class="fas fa-envelope"></i></a></div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#scvi-tools"><span class="toc-number">1.</span> <span class="toc-text">scvi-tools</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Scanorama"><span class="toc-number">2.</span> <span class="toc-text">Scanorama</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#fastMNN"><span class="toc-number">3.</span> <span class="toc-text">fastMNN</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#scGen"><span class="toc-number">4.</span> <span class="toc-text">scGen</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Harmony"><span class="toc-number">5.</span> <span class="toc-text">Harmony</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Seurat-v3"><span class="toc-number">6.</span> <span class="toc-text">Seurat v3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Seurat-v4"><span class="toc-number">7.</span> <span class="toc-text">Seurat v4</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#BBKNN"><span class="toc-number">8.</span> <span class="toc-text">BBKNN</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#ComBat"><span class="toc-number">9.</span> <span class="toc-text">ComBat</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MNN"><span class="toc-number">10.</span> <span class="toc-text">MNN</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#trVAE"><span class="toc-number">11.</span> <span class="toc-text">trVAE</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Conos"><span class="toc-number">12.</span> <span class="toc-text">Conos</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#DESC"><span class="toc-number">13.</span> <span class="toc-text">DESC</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#LIGER"><span class="toc-number">14.</span> <span class="toc-text">LIGER</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SAUCIE"><span class="toc-number">15.</span> <span class="toc-text">SAUCIE</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#scAlign"><span class="toc-number">16.</span> <span class="toc-text">scAlign</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MATCHER"><span class="toc-number">17.</span> <span class="toc-text">MATCHER</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MAGAN"><span class="toc-number">18.</span> <span class="toc-text">MAGAN</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Harmonic"><span class="toc-number">19.</span> <span class="toc-text">Harmonic</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MMD-MA"><span class="toc-number">20.</span> <span class="toc-text">MMD-MA</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SCIM"><span class="toc-number">21.</span> <span class="toc-text">SCIM</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#UnionCom"><span class="toc-number">22.</span> <span class="toc-text">UnionCom</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#coupledNMF"><span class="toc-number">23.</span> <span class="toc-text">coupledNMF</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SCOTv2"><span class="toc-number">24.</span> <span class="toc-text">SCOTv2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MOFA"><span class="toc-number">25.</span> <span class="toc-text">MOFA+</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#scmap"><span class="toc-number">26.</span> <span class="toc-text">scmap</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#BindSC"><span class="toc-number">27.</span> <span class="toc-text">BindSC</span></a></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item"><a class="thumbnail" href="/2022/01/15/scvi-tools-trap/" title="记在用scvi-tools时踩过的坑"><img src="/2022/01/15/scvi-tools-trap/scvi-tools.jpg" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="记在用scvi-tools时踩过的坑"/></a><div class="content"><a class="title" href="/2022/01/15/scvi-tools-trap/" title="记在用scvi-tools时踩过的坑">记在用scvi-tools时踩过的坑</a><time datetime="2022-01-15T03:39:38.000Z" title="发表于 2022-01-15 11:39:38">2022-01-15</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/01/12/single-cell-multiomics-integration/" title="单细胞数据整合：读Paper/翻Github时发现的一些工具"><img src="/2022/01/12/single-cell-multiomics-integration/%E5%8D%95%E7%BB%86%E8%83%9E%E6%95%B4%E5%90%88%E7%AD%96%E7%95%A5.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="单细胞数据整合：读Paper/翻Github时发现的一些工具"/></a><div class="content"><a class="title" href="/2022/01/12/single-cell-multiomics-integration/" title="单细胞数据整合：读Paper/翻Github时发现的一些工具">单细胞数据整合：读Paper/翻Github时发现的一些工具</a><time datetime="2022-01-12T06:55:17.000Z" title="发表于 2022-01-12 14:55:17">2022-01-12</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2022/01/12/biostars-first-try/" title="单细胞多模态整合：记我在Biostars上的第一次提问"><img src="/2022/01/12/biostars-first-try/biostars.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="单细胞多模态整合：记我在Biostars上的第一次提问"/></a><div class="content"><a class="title" href="/2022/01/12/biostars-first-try/" title="单细胞多模态整合：记我在Biostars上的第一次提问">单细胞多模态整合：记我在Biostars上的第一次提问</a><time datetime="2022-01-11T16:02:07.000Z" title="发表于 2022-01-12 00:02:07">2022-01-12</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2022 By Arita Zhang</div><div class="framework-info"><span>框架 </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>主题 </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="回到顶部"><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><div class="js-pjax"></div><script id="canvas_nest" defer="defer" color="0,0,255" opacity="0.7" zIndex="-1" count="99" mobile="false" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/canvas-nest.min.js"></script><script id="click-heart" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc@1/dist/click-heart.min.js" async="async" mobile="false"></script></div><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"log":false,"pluginJsPath":"lib/","pluginModelPath":"assets/","pluginRootPath":"live2dw/","tagMode":false});</script></body></html>