<!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="theme-color" content="#6200ee"><meta name="author" content="Arita Zhang"><meta name="copyright" content="Arita Zhang"><meta name="generator" content="Hexo 5.4.0"><meta name="theme" content="hexo-theme-yun"><title>单细胞数据整合：读Paper/翻Github时发现的一些工具 | Words Overflow:(</title><link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Noto+Serif+SC:wght@900&amp;display=swap" media="print" onload="this.media='all'"><link rel="stylesheet" href="https://fastly.jsdelivr.net/npm/star-markdown-css@0.3.3/dist/yun/yun-markdown.min.css"><link rel="stylesheet" href="https://fastly.jsdelivr.net/npm/prism-theme-vars/base.css"><script src="https://fastly.jsdelivr.net/npm/scrollreveal/dist/scrollreveal.min.js" defer></script><script>function initScrollReveal() {
  [".post-card",".markdown-body img"].forEach((target)=> {
    ScrollReveal().reveal(target);
  })
}
document.addEventListener("DOMContentLoaded", initScrollReveal);
document.addEventListener("pjax:success", initScrollReveal);
</script><link rel="icon" type="image/png" href="/logo2.png"><link rel="mask-icon" href="/logo2.png" color="#6200ee"><link rel="preload" href="/css/hexo-theme-yun.css" as="style"><link rel="prefetch" href="/js/sidebar.js" as="script"><link rel="preconnect" href="https://cdn.jsdelivr.net" crossorigin><link rel="preconnect" href="https://fastly.jsdelivr.net/npm/" crossorigin><script id="yun-config">
    window.Yun = {}
    window.CONFIG = {"hostname":"aritaz-hang.github.io","root":"/","title":"世界新法则的先行人","version":"1.10.4","mode":"auto","copycode":true,"page":{"isPost":true},"i18n":{"placeholder":"搜索...","empty":"找不到您查询的内容: ${query}","hits":"找到 ${hits} 条结果","hits_time":"找到 ${hits} 条结果（用时 ${time} 毫秒）"},"anonymous_image":"https://cdn.yunyoujun.cn/img/avatar/none.jpg","say":{"api":"https://v1.hitokoto.cn","hitokoto":true},"fireworks":{"colors":null},"vendors":{"darken":"https://fastly.jsdelivr.net/npm/darken@1.5.0"}};
  </script><link rel="stylesheet" href="/css/hexo-theme-yun.css"><script src="/js/hexo-theme-yun.js" type="module"></script><meta name="description" content="2022&#x2F;07&#x2F;19 Update: 其中的很多东西应该不会再写了。照着Luecken那篇Benchmarking简单的写个总结：  整合scRNA-seq数据：Scnaroma, Harmony，scVI，scGen 整合scATAC-seq数据：在Peak层面整合效果比在Gene Score&#x2F;Gene Activity层面好。在Peak层面整合可用LIGER，在Gene Score层面整合 &#x2F;">
<meta property="og:type" content="article">
<meta property="og:title" content="单细胞数据整合：读Paper&#x2F;翻Github时发现的一些工具">
<meta property="og:url" content="https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/index.html">
<meta property="og:site_name" content="Words Overflow:(">
<meta property="og:description" content="2022&#x2F;07&#x2F;19 Update: 其中的很多东西应该不会再写了。照着Luecken那篇Benchmarking简单的写个总结：  整合scRNA-seq数据：Scnaroma, Harmony，scVI，scGen 整合scATAC-seq数据：在Peak层面整合效果比在Gene Score&#x2F;Gene Activity层面好。在Peak层面整合可用LIGER，在Gene Score层面整合 &#x2F;">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2022-01-12T06:55:17.000Z">
<meta property="article:modified_time" content="2022-07-19T11:42:29.736Z">
<meta property="article:author" content="Arita Zhang">
<meta property="article:tag" content="单细胞数据整合">
<meta property="article:tag" content="单细胞工具">
<meta name="twitter:card" content="summary"><script>(function() {
  const prefersDark = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
  const setting = localStorage.getItem('darken-mode') || 'auto'
  if (setting === 'dark' || (prefersDark && setting !== 'light'))
    document.documentElement.classList.toggle('dark', true)
})()</script></head><body><script src="https://code.iconify.design/2/2.1.1/iconify.min.js"></script><script>// Define global variable
IconifyProviders = {
  // Empty prefix: overwrite default API provider configuration
  '': {
    // Use custom API first, use Iconify public API as backup
    resources: [
        'https://api.iconify.design',
    ],
    // Wait for 1 second before switching API hosts
    rotate: 1000,
  },
};</script><script defer src="https://fastly.jsdelivr.net/npm/animejs@latest"></script><script defer src="/js/ui/fireworks.js" type="module"></script><canvas class="fireworks"></canvas><div class="container"><a class="sidebar-toggle hty-icon-button" id="menu-btn"><div class="hamburger hamburger--spin" type="button"><span class="hamburger-box"><span class="hamburger-inner"></span></span></div></a><div class="sidebar-toggle sidebar-overlay"></div><aside class="sidebar"><script src="/js/sidebar.js" type="module"></script><ul class="sidebar-nav"><li class="sidebar-nav-item sidebar-nav-toc hty-icon-button sidebar-nav-active" data-target="post-toc-wrap" title="文章目录"><span class="icon iconify" data-icon="ri:list-ordered"></span></li><li class="sidebar-nav-item sidebar-nav-overview hty-icon-button" data-target="site-overview-wrap" title="站点概览"><span class="icon iconify" data-icon="ri:passport-line"></span></li></ul><div class="sidebar-panel" id="site-overview-wrap"><div class="site-info fix-top"><a class="site-author-avatar" href="/about/" title="Arita Zhang"><img width="96" loading="lazy" src="/images/avatar.png" alt="Arita Zhang"><span class="site-author-status" title="不想上班">😭</span></a><div class="site-author-name"><a href="/about/">Arita Zhang</a></div><span class="site-name">Words Overflow:(</span><sub class="site-subtitle">欢迎来到Arita的个站!</sub><div class="site-description">-流而不盈 持中守正-</div></div><nav class="site-state"><a class="site-state-item hty-icon-button icon-home" href="/" title="首页"><span class="site-state-item-icon"><span class="icon iconify" data-icon="ri:home-4-line"></span></span></a><div class="site-state-item"><a href="/archives/" title="归档"><span class="site-state-item-icon"><span class="icon iconify" data-icon="ri:archive-line"></span></span><span class="site-state-item-count">4</span></a></div><div class="site-state-item"><a href="/categories/" title="分类"><span class="site-state-item-icon"><span class="icon iconify" data-icon="ri:folder-2-line"></span></span><span class="site-state-item-count">2</span></a></div><div class="site-state-item"><a href="/tags/" title="标签"><span class="site-state-item-icon"><span class="icon iconify" data-icon="ri:price-tag-3-line"></span></span><span class="site-state-item-count">4</span></a></div><a class="site-state-item hty-icon-button" target="_blank" rel="noopener" href="https://yun.yunyoujun.cn" title="文档"><span class="site-state-item-icon"><span class="icon iconify" data-icon="ri:settings-line"></span></span></a></nav><hr style="margin-bottom:0.5rem"><hr style="margin:0.5rem 1rem"><div class="links"><a class="links-item hty-icon-button" href="/links/" title="我的小伙伴们" style="color:dodgerblue"><span class="icon iconify" data-icon="ri:genderless-line"></span></a></div><br><a class="links-item hty-icon-button" id="toggle-mode-btn" href="javascript:;" title="Mode" style="color: #f1cb64"><span class="icon iconify" data-icon="ri:contrast-2-line"></span></a></div><div class="sidebar-panel sidebar-panel-active" id="post-toc-wrap"><div class="post-toc"><div class="post-toc-content"><ol class="toc"><li class="toc-item toc-level-2"><a class="toc-link" href="#scvi-tools"><span class="toc-number">1.</span> <span class="toc-text">scvi-tools</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Scanorama"><span class="toc-number">2.</span> <span class="toc-text">Scanorama</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#fastMNN"><span class="toc-number">3.</span> <span class="toc-text">fastMNN</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#scGen"><span class="toc-number">4.</span> <span class="toc-text">scGen</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Harmony"><span class="toc-number">5.</span> <span class="toc-text">Harmony</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Seurat-v3"><span class="toc-number">6.</span> <span class="toc-text">Seurat v3</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Seurat-v4"><span class="toc-number">7.</span> <span class="toc-text">Seurat v4</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#BBKNN"><span class="toc-number">8.</span> <span class="toc-text">BBKNN</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#ComBat"><span class="toc-number">9.</span> <span class="toc-text">ComBat</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MNN"><span class="toc-number">10.</span> <span class="toc-text">MNN</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#trVAE"><span class="toc-number">11.</span> <span class="toc-text">trVAE</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Conos"><span class="toc-number">12.</span> <span class="toc-text">Conos</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#DESC"><span class="toc-number">13.</span> <span class="toc-text">DESC</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#LIGER"><span class="toc-number">14.</span> <span class="toc-text">LIGER</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SAUCIE"><span class="toc-number">15.</span> <span class="toc-text">SAUCIE</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#scAlign"><span class="toc-number">16.</span> <span class="toc-text">scAlign</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MATCHER"><span class="toc-number">17.</span> <span class="toc-text">MATCHER</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MAGAN"><span class="toc-number">18.</span> <span class="toc-text">MAGAN</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Harmonic"><span class="toc-number">19.</span> <span class="toc-text">Harmonic</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MMD-MA"><span class="toc-number">20.</span> <span class="toc-text">MMD-MA</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SCIM"><span class="toc-number">21.</span> <span class="toc-text">SCIM</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#UnionCom"><span class="toc-number">22.</span> <span class="toc-text">UnionCom</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#coupledNMF"><span class="toc-number">23.</span> <span class="toc-text">coupledNMF</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#SCOTv2"><span class="toc-number">24.</span> <span class="toc-text">SCOTv2</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#MOFA"><span class="toc-number">25.</span> <span class="toc-text">MOFA+</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#scmap"><span class="toc-number">26.</span> <span class="toc-text">scmap</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#BindSC"><span class="toc-number">27.</span> <span class="toc-text">BindSC</span></a></li></ol></div></div></div></aside><main class="sidebar-translate" id="content"><div id="post"><article class="hty-card post-block" itemscope itemtype="https://schema.org/Article" style="--smc-primary:#6200ee;"><link itemprop="mainEntityOfPage" href="https://AritaZ-hang.github.io/2022/01/12/single-cell-multiomics-integration/"><span hidden itemprop="author" itemscope itemtype="https://schema.org/Person"><meta itemprop="name" content="Arita Zhang"><meta itemprop="description"></span><span hidden itemprop="publisher" itemscope itemtype="https://schema.org/Organization"><meta itemprop="name" content="Words Overflow:("></span><header class="post-header"><h1 class="post-title" itemprop="name headline">单细胞数据整合：读Paper/翻Github时发现的一些工具</h1><div class="post-meta"><div class="post-time"><span class="post-meta-item-icon"><span class="icon iconify" data-icon="ri:calendar-line"></span></span> <time title="创建时间：2022-01-12 14:55:17" itemprop="dateCreated datePublished" datetime="2022-01-12T14:55:17+08:00">2022-01-12</time><span class="post-meta-divider">-</span><span class="post-meta-item-icon"><span class="icon iconify" data-icon="ri:calendar-2-line"></span></span> <time title="修改时间：2022-07-19 19:42:29" itemprop="dateModified" datetime="2022-07-19T19:42:29+08:00">2022-07-19</time></div><div class="post-classify"><span class="post-category"> <span class="post-meta-item-icon" style="margin-right:3px;"><span class="icon iconify" data-icon="ri:folder-line"></span></span><span itemprop="about" itemscope itemtype="https://schema.org/Thing"><a class="category-item" href="/categories/%E5%8D%95%E7%BB%86%E8%83%9E%E5%88%86%E6%9E%90/" style="--text-color:var(--hty-text-color)" itemprop="url" rel="index"><span itemprop="text">单细胞分析</span></a></span></span><span class="post-tag"><span class="post-meta-divider">-</span><a class="tag-item" href="/tags/%E5%8D%95%E7%BB%86%E8%83%9E%E6%95%B0%E6%8D%AE%E6%95%B4%E5%90%88/" style="--text-color:var(--hty-text-color)"><span class="post-meta-item-icon"><span class="icon iconify" data-icon="ri:price-tag-3-line"></span></span><span class="tag-name">单细胞数据整合</span></a><a class="tag-item" href="/tags/%E5%8D%95%E7%BB%86%E8%83%9E%E5%B7%A5%E5%85%B7/" style="--text-color:var(--hty-text-color)"><span class="post-meta-item-icon"><span class="icon iconify" data-icon="ri:price-tag-3-line"></span></span><span class="tag-name">单细胞工具</span></a></span></div></div></header><section class="post-body" itemprop="articleBody"><div class="post-content markdown-body"><p><strong>2022/07/19 Update</strong>:</p>
<p>其中的很多东西应该不会再写了。照着Luecken那篇Benchmarking简单的写个总结：</p>
<ol>
<li>整合scRNA-seq数据：Scnaroma, Harmony，scVI，scGen</li>
<li>整合scATAC-seq数据：在Peak层面整合效果比在Gene Score/Gene Activity层面好。在Peak层面整合可用LIGER，在Gene Score层面整合 /需要结合scRNA-seq数据的话，scVI/scANVI是个比较好的选择。</li>
</ol>
<hr>
<p><strong>ATTENTION:</strong> 本文的内容将持续更新。</p>
<p>总结一下近期读paper/翻Github时发现的一些可以用于做单细胞数据整合的工具，大致分为去除批次效应 、单模态数据整合、多模态数据整合几个大类，一些能用在单一模态（如scRNA-seq）上的工具因为我本人没用过不清楚能不能迁移到其他模态上（如染色质可及性、DNA甲基化等），在适用范围上也就先不标了。</p>
<p>Paper主要参考的是Luecken等人22年发在Nature Method上的Benchmarking、Argelaguet等人21年发在Nature Biotechnology上的综述，以及各个工具的作者在其文章Introduction &amp; Results部分提及的其他工具的算法原理与性能情况。</p>
<p>然后再贴一下awesome-single-cell这个Github Repository，单细胞数据分析软件贴的比较全，整合算法因为最近一两年出来的比较多上面还没怎么更新。<a target="_blank" rel="noopener" href="https://github.com/seandavi/awesome-single-cell">Github Link</a>。</p>
<p><em>WARNING:</em> 因为本人对算法原理都不是很熟悉，且数学知识早就还给教过我的各位老师了，尽管描述算法的部分写的都比较简陋，还是看看就好OTL</p>
<p>以下↓</p>
<ul>
<li><h2 id="scvi-tools"><a href="#scvi-tools" class="headerlink" title="scvi-tools"></a><a target="_blank" rel="noopener" href="https://docs.scvi-tools.org/en/stable/index.html">scvi-tools</a></h2><ol>
<li><p>18年至今陆陆续续开发的一组工具包，应用<code>神经网络</code>基于<code>潜在变量模型</code>（Latent Variable models）开发深度生成模型（DGMs）解决诸如降维、聚类、差异表达分析、注释、去除不想要的变异（去批次效应） &amp; 跨模态整合的问题。该套工具包包括可分析scRNA-seq的工具（scVI、scArches），单细胞注释（CellAssign, scANVI），空间转录组数据的去卷积化（Stereoscope, DestVI），去除双细胞（Solo），基于CITE-seq数据的多模态分析（totalVI），染色质可及性分析（PeakVI）等。</p>
<blockquote>
<p>还是直接抄我本科毕业论文上对这东西的介绍吧：</p>
<p>scANVI是一种基于变分自编码算法（Variational Auto Encoder）和神经网络的深度生成模型，其实现细胞类型注释标签转移的过程主要包含以下几步：</p>
<p>（1）基于零膨胀负二项分布（Zero-inflated negative binomial，ZINB）对每个细胞的基因表达水平进行归一化，使用变分自编码算法将高维的基因表达投射到低维，计算每个细胞在共同的低维潜在空间的概率表示，进一步得到参考细胞类型的分类分布。 </p>
<p>（2）基于KNN算法捕捉共同低维潜在空间中细胞间的邻近性（Proximity），为每个查询（Query）细胞匹配与之表达最相似的参考（Reference）细胞。</p>
<p>（3）利用半监督贝叶斯算法和深度神经网络计算变分模型的参数，估计每个查询细胞在各参考类型分类分布上的后验概率，最终匹配细胞和细胞类型注释结果，实现标签转移。</p>
</blockquote>
</li>
<li><p>scvi-tools利用AnnData数据结构，可和Scanpy集成；同时也具有R接口（reticulate），在一定程度上可以和Seurat/Bioconductor协同工作。</p>
</li>
<li><p>利用神经网络进行训练需要一定的电脑配置（无现成GPU资源的情况下推荐用Google的colab白嫖GPU），大数据（几十k细胞）用CPU训练模型极其缓慢。</p>
<blockquote>
<p>简单照着流程跑了一下，scANVI的integration - label transfer效果还是相当不错的，利用Ground Truth检验，具有组织特异性的细胞的预测正确率基本可以达到95%以上；但在对多组织的数据进行操作时，对cross-tissues的血细胞等预测效果不甚理想，可以考虑将组织进行拆分进行1-1映射，应该可以在一定程度上提高预测效果。</p>
</blockquote>
</li>
</ol>
</li>
<li><h2 id="Scanorama"><a href="#Scanorama" class="headerlink" title="Scanorama"></a><a target="_blank" rel="noopener" href="http://cb.csail.mit.edu/cb/scanorama/">Scanorama</a></h2><ol>
<li>18年发布的一款可去除scRNA-seq批次效应 &amp; 整合数据的算法，利用<code>全景拼接</code>方法，根据两个数据集Overlap的区域进行batch-correction和Integration，能够保留数据 specific population，也不需要所有的dataset share相同的cell type。</li>
<li>利用mutual nearest neighbors matching ，主要能够寻找两个数据集的相似部分，而不是整个大数据中的相似部分。针对多于两个datasets的比对，先选择一个作为reference，之后将其他的datasets和这个datasets做integrate，所以整合的效果很看重数据的顺序。</li>
<li>Scanorama 的两个关键性步骤：1.使用SVD（singular value decomposition） 将高维的数据降维。2.基于超敏感局部敏感散列和随机投影树来构建近邻法，节约时间。</li>
<li>主要适用于scRNA-seq的批次效应去除 &amp; 整合，在Argelaguet的综述中被归类于Horizontal-Integration green。</li>
<li>Python package，利用AnnData作为数据结构，可集成Scanpy。</li>
</ol>
</li>
<li><h2 id="fastMNN"><a href="#fastMNN" class="headerlink" title="fastMNN"></a>fastMNN</h2></li>
<li><h2 id="scGen"><a href="#scGen" class="headerlink" title="scGen"></a><a target="_blank" rel="noopener" href="https://scgen.readthedocs.io/en/latest/">scGen</a></h2><ol>
<li>基于结合向量算法的VAE，具有适合scRNA-seq数据的体系结构。scGen能够预测细胞在药剂或感染下的变化，以预测细胞类型；且scGen能够跨物种的挖掘细胞受扰动产生的表达改变。</li>
<li>（从scGen的原始论文来看），scGen适用于<strong>scRNA-seq对照组-实验组</strong>的去批次效应和整合，也适合做跨物种分析，是否适用于更广范围的去批次效应 &amp; scRNA-seq整合需要进一步实验。</li>
<li>Python package, 利用AnnData为数据结构，也可集成Scanpy（scGen的去批次效应和分析需要scanpy的<code>highly_variable_gene</code>函数。</li>
</ol>
</li>
<li><h2 id="Harmony"><a href="#Harmony" class="headerlink" title="Harmony"></a>Harmony</h2><ol>
<li>scRNA-seq &amp; scATAC-seq的数据整合都可以用这个东西。在Luecken那篇Benchmarking里，在Horizontal整合scRNA-seq数据时有前三的表现。至于scATAC-seq数据整合，ArchR设计了一个可以调用Harmony的接口，总的来说还是挺好用的。</li>
</ol>
</li>
<li><h2 id="Seurat-v3"><a href="#Seurat-v3" class="headerlink" title="Seurat v3"></a>Seurat v3</h2><ol>
<li>老牌scRNA-seq分析工具，v3版本似乎可直接根据scATAC-seq的染色质峰矩阵计算Gene Activity进行scRNA-seq &amp; scATAC-seq的整合，在Argelaguet的综述中被归类为Horizontal-Integration。</li>
<li>虽然没有用过，但在Leucken的Benchmarking文章中去批次效应结果&amp;整合结果都表现欠佳。总之和它的后续版本Seurat v4一样，如果能选择基于深度学习的工具，那就不要用它做整合。</li>
<li>这东西是个R包就不用说了吧。</li>
</ol>
</li>
<li><h2 id="Seurat-v4"><a href="#Seurat-v4" class="headerlink" title="Seurat v4"></a><a target="_blank" rel="noopener" href="https://satijalab.org/seurat/index.html">Seurat v4</a></h2><ol>
<li>老牌scRNA-seq分析工具，v4版本在代码风格上和v3版本有一定差异；该版本结合了该Satija实验室开发的Signac工具，可将scATAC-seq的fragments数据转化为基因活度（Gene Activity）。v4版本的Seurat利用CCA算法将基因表达 &amp; 基因活度降维后寻找两种模态中通用的anchor进行scRNA-seq -&gt; scATAC-seq的映射，从而完成两种模态的整合。</li>
<li>集成了WNN算法，对单细胞多模态数据（<strong>profiled in same cells</strong>）的矫正&amp;整合力度似乎还不错，在Argelaguet的综述中被归类为Vertical-Integration。</li>
<li>只能进行scRNA-seq和scATAC-seq的整合；采用的anchors算法在挖掘两种模态的通用特征上不够深入；将染色质可及性的峰矩阵转换为基因活度矩阵会丢失一部分原数据的特性，进一步影响效果。</li>
<li>多个比较实验中，Seurat v4在标签转移、多模态数据整合上表现欠佳（见Leucken/UnionCom/SCOTv2文章）。&amp; 一点个人经验：如果能用基于深度学习的工具（SCOT、scVI、UnionCom等），那就不要用Seurat v4。</li>
<li>这东西是个R包就不用说了吧。</li>
</ol>
</li>
<li><h2 id="BBKNN"><a href="#BBKNN" class="headerlink" title="BBKNN"></a>BBKNN</h2></li>
<li><h2 id="ComBat"><a href="#ComBat" class="headerlink" title="ComBat"></a>ComBat</h2></li>
<li><h2 id="MNN"><a href="#MNN" class="headerlink" title="MNN"></a>MNN</h2></li>
<li><h2 id="trVAE"><a href="#trVAE" class="headerlink" title="trVAE"></a>trVAE</h2></li>
<li><h2 id="Conos"><a href="#Conos" class="headerlink" title="Conos"></a>Conos</h2></li>
<li><h2 id="DESC"><a href="#DESC" class="headerlink" title="DESC"></a>DESC</h2></li>
<li><h2 id="LIGER"><a href="#LIGER" class="headerlink" title="LIGER"></a>LIGER</h2></li>
<li><h2 id="SAUCIE"><a href="#SAUCIE" class="headerlink" title="SAUCIE"></a>SAUCIE</h2></li>
<li><h2 id="scAlign"><a href="#scAlign" class="headerlink" title="scAlign"></a>scAlign</h2></li>
<li><h2 id="MATCHER"><a href="#MATCHER" class="headerlink" title="MATCHER"></a>MATCHER</h2></li>
<li><h2 id="MAGAN"><a href="#MAGAN" class="headerlink" title="MAGAN"></a>MAGAN</h2></li>
<li><h2 id="Harmonic"><a href="#Harmonic" class="headerlink" title="Harmonic"></a>Harmonic</h2></li>
<li><h2 id="MMD-MA"><a href="#MMD-MA" class="headerlink" title="MMD-MA"></a>MMD-MA</h2></li>
<li><h2 id="SCIM"><a href="#SCIM" class="headerlink" title="SCIM"></a>SCIM</h2><ol>
<li>SCIM假设细胞共享一个共同的（低维的）底层结构，并且底层细胞的分布在不同的技术中心是近似恒定的，利用一个具有对抗性目标的自动编码器框架构造了一个技术不变的潜在空间。多模态数据集市通过配对跨技术的单元来集成的，使用一个二部匹配方案，操作在低维潜在表示。</li>
<li>主要应用于分别测定的单细胞多模态数据（<strong>profiled in different cells</strong>）的整合，在Argelaguet的综述中被归类为Diagonal-Integration green。</li>
</ol>
</li>
<li><h2 id="UnionCom"><a href="#UnionCom" class="headerlink" title="UnionCom"></a><a target="_blank" rel="noopener" href="https://github.com/caokai1073/UnionCom">UnionCom</a></h2><ol>
<li>一种基于流型策略，利用了拓扑比对（Topological Alignment）的无监督算法。UnionCom将每个单细胞数据集固有的低维结构嵌入到几何距离矩阵中，通过匹配几何距离矩阵，跨单细胞多组学数据集对齐细胞，最终将不同的不对应特征投射到一个共同的embedding空间中。</li>
<li>主要应用于分别测定的单细胞多模态数据(<strong>profiled in different cells</strong>)的整合，Diagonal-Integration。</li>
<li>不需要事先的细胞关联信息作为输入（不需要各模态所检测细胞的1-1匹配，也不需要事先的Cell Type Labels）；采用非线性方法，在非线性特征（如染色质可及性上）较线性方法如Seurat/MNN/Harmony等有更好的性能。</li>
<li>是<strong>Python</strong>工具包，可以在代码框中运行，不依赖Linux命令行。</li>
<li>基于UnionCom，作者Cao Kai开发了<a target="_blank" rel="noopener" href="https://github.com/caokai1073/Pamona">Panoma</a>，可集成细胞类型的先验信息。</li>
</ol>
</li>
<li><h2 id="coupledNMF"><a href="#coupledNMF" class="headerlink" title="coupledNMF"></a>coupledNMF</h2></li>
<li><h2 id="SCOTv2"><a href="#SCOTv2" class="headerlink" title="SCOTv2"></a><a target="_blank" rel="noopener" href="https://github.com/rsinghlab/SCOT">SCOTv2</a></h2><ol>
<li>一种基于不平衡最佳传输（Unbalanced Optimal transport）策略的无监督算法，该v2版本算法在SCOT（采用Gromov-Wasserstein最佳传输算法）的基础上增加了散度项，以允许边缘值的质量变化。为允许更多模态（N&gt;2）的整合，使用分类更为清晰质量更好的domain（我猜测可能是整个模态的数据）作为anchor进行不同模态间的锚定。SCOTv2采用了UnionCom利用的调整后的tsne，基于偶联的细胞-细胞转移矩阵（cell-cell transition matrix）进行coembedding的展示。</li>
<li>主要应用于分别测定的单细胞多模态数据（<strong>profiled in different cells</strong>）的整合，可适用于细胞类型不平衡、缺乏1-1细胞关联的情况，属于Diagonal-Integration 。</li>
<li>该算法可以在构建模型的过程中自动选取最优参数。</li>
<li>可一次性整合多个模态（N&gt;2，例如可一次性整合scNMT-seq的基因表达、染色质可及性、DNA甲基化到一张图中），且不需要事先的Cell Type Labels暨分类依据作为输入。</li>
<li>可直接在CPU上运行，在多种算法中运行速度较快；但其对内存的需求为$O(N^2)$，可能不适用于大规模的数据。</li>
<li>是<strong>Python</strong>工具包，可以在代码框中运行，不依赖Linux命令行（这篇文章中使用的电脑配置简直就是贫民款）。</li>
</ol>
</li>
<li><h2 id="MOFA"><a href="#MOFA" class="headerlink" title="MOFA+"></a>MOFA+</h2></li>
<li><h2 id="scmap"><a href="#scmap" class="headerlink" title="scmap"></a>scmap</h2></li>
<li><h2 id="BindSC"><a href="#BindSC" class="headerlink" title="BindSC"></a>BindSC</h2><ol>
<li>通过将scATAC-seq的峰矩阵转化为基因活度矩阵，进行scRNA-seq &amp; scATAC-seq的整合。</li>
<li>在SCOTv2的论文中，在label transfer &amp; integration embedding上表现不佳，如果条件允许还是建议选择其他的工具进行整合。</li>
</ol>
</li>
</ul>
</div></section><div id="reward-container"><span class="hty-icon-button button-glow" id="reward-button" title="打赏" onclick="var qr = document.getElementById(&quot;qr&quot;); qr.style.display = (qr.style.display === &quot;none&quot;) ? &quot;block&quot; : &quot;none&quot;;"><span class="icon iconify" data-icon="ri:hand-coin-line"></span></span><div id="reward-comment">I'm so cute. Please give me money.</div></div><ul class="post-copyright"><li class="post-copyright-author"><strong>本文作者：</strong>Arita Zhang</li><li class="post-copyright-link"><strong>本文链接：</strong><a href="https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/" title="单细胞数据整合：读Paper/翻Github时发现的一些工具">https://aritaz-hang.github.io/2022/01/12/single-cell-multiomics-integration/</a></li><li class="post-copyright-license"><strong>版权声明：</strong>本博客所有文章除特别声明外，均默认采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/deed.zh" target="_blank" rel="noopener" title="CC BY-NC-SA 4.0 "><span class="icon iconify" data-icon="ri:creative-commons-line"></span><span class="icon iconify" data-icon="ri:creative-commons-by-line"></span><span class="icon iconify" data-icon="ri:creative-commons-nc-line"></span><span class="icon iconify" data-icon="ri:creative-commons-sa-line"></span></a> 许可协议。</li></ul></article><div class="post-nav"><div class="post-nav-item"><a class="post-nav-prev" href="/2022/01/15/scvi-tools-trap/" rel="prev" title="记在用scvi-tools时踩过的坑"><span class="icon iconify" data-icon="ri:arrow-left-s-line"></span><span class="post-nav-text">记在用scvi-tools时踩过的坑</span></a></div><div class="post-nav-item"><a class="post-nav-next" href="/2022/01/12/biostars-first-try/" rel="next" title="单细胞多组学整合：第一次接触BioStars"><span class="post-nav-text">单细胞多组学整合：第一次接触BioStars</span><span class="icon iconify" data-icon="ri:arrow-right-s-line"></span></a></div></div></div><div class="hty-card" id="comment"></div></main><footer class="sidebar-translate" id="footer"><div class="copyright"><span>&copy; 2019 – 2022 </span><span class="with-love" id="animate"><span class="icon iconify" data-icon="ri:cloud-line"></span></span><span class="author"> Arita Zhang</span></div><div class="powered"><span>由 <a href="https://hexo.io" target="_blank" rel="noopener">Hexo</a> 驱动 v5.4.0</span><span class="footer-separator">|</span><span>主题 - <a rel="noopener" href="https://github.com/YunYouJun/hexo-theme-yun" target="_blank"><span>Yun</span></a> v1.10.4</span></div></footer></div><a class="hty-icon-button" id="back-to-top" aria-label="back-to-top" href="#"><span class="icon iconify" data-icon="ri:arrow-up-s-line"></span><svg class="progress-circle-container" viewBox="0 0 100 100"><circle class="progress-circle" id="progressCircle" cx="50" cy="50" r="48" fill="none" stroke="#6200ee" stroke-width="2" stroke-linecap="round"></circle></svg></a><script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"log":false,"pluginJsPath":"lib/","pluginModelPath":"assets/","pluginRootPath":"live2dw/","tagMode":false});</script></body></html>